### ISSUE TITLE
Volume control is not logarithmic

### ISSUE THREAD

**Issue by ts1** [ExistentialAudio/BlackHole#149]
Hi, I tried BlackHole and found the volume control is not very useful.
The auditory change from about 0.1 to 1 is very gradual, as opposed to the change of 0 to about 0.1.

From looking at the source code, the dB to linear volume conversion is not correct and is implemented as a linear conversion from kVolume_MinDB to kVolume_MaxDB.

As I understand it, the linear volume should be calculated as `pow(10.0, dB/20.0)`, and the individual samples should be multiplied by that value.

"scalar" values are for the UI, the lowest dB value should be determined to be -60dB for example, and any non-zero value should correspond linearly to the dB value. (0 is 0 in linear volume and -∞ in dB)

To sum it up, there should be three volume values.
- Linear volume (my term) - the value to be multiplied to the individual samples
- dB volume - `20 * log10(linear volume)`
- UI ("scalar") volume - a linear representation of the lowest dB value to 0 dB

**Comment by ts1** [comment]
On closer examination, I found that setting -60dB as the lowest value would bias the volume too much towards the low end. On the other hand, losing control of the lower areas is also a problem.
This is a deep problem, and the current squared implementation seems like a not so bad solution (although the dB indication is not correct).
Reference: https://www.dr-lex.be/info-stuff/volumecontrols.html

I'll try to experiment with it myself when I can.

**Comment by devinroth** [comment]
Shoot. I didn’t think anyone would notice. :) 

I’m happy to get received any input.

**Comment by devinroth** [comment]
Sweet thanks for the changes.

**Comment by ts1** [comment]
Thank you for merging!
